tests/unit/ct-completion.test.ts

```diff
@@ -0,0 +1,118 @@
+import { describe, expect, test, vi, afterEach, beforeEach } from "vitest";
+import type * as PythonUtilities from "../../server/pythonUtilities";
+import { getCompletionPositionForLine, generateCompletionData } from "./utils";
+
+const unitTst = `
+// ---------------------------------------------------------------------------------------
+// Simple Example - new
+// vmock vmock_examples simpleFunction 
+int vmock_vmock_examples_simpleFunction(::vunit::CallCtx<> vunit_ctx, char param1, float param2) {
+  // Enable Stub: vmock_vmock_examples_simpleFunction_enable_disable(vmock_session);
+  // Disable Stub: vmock_vmock_examples_simpleFunction_enable_disable(vmock_session, false);
+
+  return 100;
+}
+void vmock_vmock_examples_simpleFunction_enable_disable(vunit::MockSession &vmock_session, bool enable = true) {
+    using vcast_mock_rtype = int ;
+    vcast_mock_rtype (*vcast_fn_ptr)(char param1, float param2)  = &simpleFunction;
+    vmock_session.mock <vcast_mock_rtype (*)(char param1, float param2)> ((vcast_mock_rtype (*)(char param1, float param2))vcast_fn_ptr).assign (enable ? &vmock_vmock_examples_simpleFunction : nullptr);
+}
+
+VTEST(vmockExamples, simpleTest2) {
+
+  auto vmock_session = ::vunit::MockSession();
+  vmock_vmock_examples_simpleFunction_enable_disable(vmock_session);
+  VASSERT_EQ (100, simpleFunction ('a', 1.0));
+
+  // disable the stub, which means the real code will return param1 'a' or 97
+  vmock_vmock_examples_simpleFunction_enable_disable(vmock_session, false);
+  VASSERT_EQ (97, simpleFunction ('a', 1.0));
+
+}
+`;
+
+const timeout = 30_000; // 30 seconds
+
+describe("Testing pythonUtilities (valid)", () => {
+  beforeEach(() => {
+    // To achieve 100% coverage
+    // --> Need to mock getChoiceDataFromPython, since "extraText" is not implemented yet
+    vi.mock("../../server/pythonUtilities", async () => {
```

Comment:
I don't think any other test mocks this stuff; I think we want it to run the _real_ code and get the completion stuff back.

Reply:
Added one test without mocking and extraText in 3c5d87311155ac78144ae54462bfa86fb762002d .